index.ts

```diff
@@ -2,61 +2,31 @@ import { version } from './package.json';
 
 const externalStaticOrigin = 'https://javascript.wert.io';
 
-interface options {
-  partner_id?: string
-  origin?: string
-  address?: string
-  theme?: string
-  currency?: string
-  currency_amount?: number
-  commodity?: string
-  commodity_amount?: number
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [x: string]: any
-}
-
-interface extraOptions {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [x: string]: any
-}
-
-interface eventOptions {
-  type: string
-  origin: string
-  data: {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    [x: string]: any
-  }
-}
-
-type customListener = (data: {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [x: string]: any
-}) => void;
-
-type setThemeData = {
-  theme?: string
-  colors?: {
-    [x: string]: string
-  }
-};
-
+import type {
+  GivenOptions,
+  WidgetOptions,
+  ExtraOptions,
+  EventTypes,
+  OptionalEventTypes,
+  EventListeners,
+  CustomListener,
+  WidgetEvent,
+  SendEventParameters,
+  SetThemeParameters,
+} from './types';
 class WertWidget {
-
   private iframe: HTMLIFrameElement = document.createElement('iframe');
 
   partner_id?: string;
```

Comment:
We actually get straight to the error screen in the widget if we don't pass the partner_id. However, I don't think we can make it required, because we'll have to add the 'default' value to it then, and it doesn't seem right to me. What would you suggest?